### PR TITLE
on failure to add network trying to use exisiting configuration, #36

### PR DESCRIPTION
**Pull request by CityVibes**
If adding network fails setup looks for existing configuration by SSID and tries to use that instead. While testing solution reduced failure rate on nougat significantly.

### CODE REVIEW

File: devicesetup/src/main/java/io/particle/android/sdk/devicesetup/ApConnector.java [hunk 1]
```diff
@@ -147,6 +146,14 @@ public SSID connectToAP(final WifiConfiguration config, Client client) {
             setupRunnables.add(() -> {
                 log.d("Adding network " + configSSID);
                 networkID.set(wifiFacade.addNetwork(config));
+
+                if (networkID.get() == -1) {
```
Comment (jensck) [inline devicesetup/src/main/java/io/particle/android/sdk/devicesetup/ApConnector.java]: So we might have been getting back `-1` even when it succeeded?

**Comment by jensck** [comment]
@CityVibes have you tested this on a Samsung device yet?

File: devicesetup/src/main/java/io/particle/android/sdk/devicesetup/ApConnector.java (hunk 1 above)
Comment (CityVibes) [inline devicesetup/src/main/java/io/particle/android/sdk/devicesetup/ApConnector.java]: Sometimes we used to get `-1` on `addNetwork` in Nougat, not really succeeding when it fails to add network.

**Comment by CityVibes** [comment]
@jensck I have samsung S7, so tested on that.

File: devicesetup/src/main/java/io/particle/android/sdk/devicesetup/ApConnector.java (hunk 1 above)
Comment (jensck) [inline devicesetup/src/main/java/io/particle/android/sdk/devicesetup/ApConnector.java]: Not understanding your reply, sorry. :)

Seems like one of these things must be true:
1. `addNetwork()` added the network, but reported failure (-1), or...
2. `removeNetwork()` failed (which then caused `addNetwork()` to fail), and we're just using the previous config?

Did your testing suggest which of these it is?

**Comment by jensck** [comment]
Excellent.

You can merge when you're ready, I just want to get clear on what was happening with the network add/remove functions.

File: devicesetup/src/main/java/io/particle/android/sdk/devicesetup/ApConnector.java (hunk 1 above)
Comment (CityVibes) [inline devicesetup/src/main/java/io/particle/android/sdk/devicesetup/ApConnector.java]: It's "addNetwork() added the network, but reported failure (-1)", removing network did not report failure.

**Pull request merged**